adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/authorization/PolicyEnforcer.java

```diff
@@ -263,20 +267,35 @@ public PathConfig matches(String targetUri) {
             pathConfig = super.matches(targetUri);
 
             if (enforcerConfig.getLazyLoadPaths() || enforcerConfig.getPathCacheConfig() != null) {
-                if ((pathConfig == null || (pathConfig.getPath().contains("*")))) {
+                if ((pathConfig == null || pathConfig.isInvalidated() || pathConfig.getPath().contains("*"))) {
                     try {
                         List<ResourceRepresentation> matchingResources = authzClient.protection().resource().findByMatchingUri(targetUri);
 
-                        if (!matchingResources.isEmpty()) {
+                        if (matchingResources.isEmpty()) {
+                            // if this config is invalidated (e.g.: due to cache expiration) we remove and return null
+                            if (pathConfig != null && pathConfig.isInvalidated()) {
+                                paths.remove(targetUri);
+                                return null;
+                            }
+                        } else {
                             Map<String, Map<String, Object>> cipConfig = null;
                             PolicyEnforcerConfig.EnforcementMode enforcementMode = PolicyEnforcerConfig.EnforcementMode.ENFORCING;
+                            ResourceRepresentation targetResource = matchingResources.get(0);
 
                             if (pathConfig != null) {
                                 cipConfig = pathConfig.getClaimInformationPointConfig();
                                 enforcementMode = pathConfig.getEnforcementMode();
+                            } else {
+                                for (PathConfig existingPath : paths.values()) {
+                                    if (existingPath.getId().equals(targetResource.getId()) 
+                                            && existingPath.isStatic()
+                                            && !PolicyEnforcerConfig.EnforcementMode.DISABLED.equals(existingPath.getEnforcementMode())) {
```

Comment:
Could you please elaborate a little bit more on why is this condition necessary? I would understand it without the exclamation mark in the last condition, but not with it. Maybe in some example situation when this happens.

Reply:
If the path definition is not explicitly disabled in the configuration (enforcing or permissive) then we go with the invalidation. Otherwise, paths disabled in the configuration are going to be overridden when re-fetched and not be disabled anymore.